pkg/module/resources.go

```diff
@@ -117,7 +118,7 @@ func addBlob(fs vfs.FileSystem, archive *comparch.ComponentArchive, resource *Re
 		return err
 	}
 
-	return archive.SetResource(&resource.ResourceMeta, blobAccess)
+	return archive.SetResource(&resource.ResourceMeta, blobAccess, cpi.ModifyResource(true))
```

Comment:
The answer is: I don't know (yet). Without it, the current code fails with a message (I am recalling from my memory, these aren't exact words): "changes to archive would invalidate the digest signature"
It may be that the way we're constructing the component archive is not "optimal" - it is different from what OCM library authors assume as a "right path". And once we change our code, this option is no longer necessary.

Reply:
I reproduced it with same error